### PR TITLE
Catavatar: The profile picture is now stored in a better way

### PR DESCRIPTION
**Pull request by annando**
Meow!

### CODE REVIEW

File: catavatar/catavatar.php [hunk 1]
```diff
@@ -92,36 +95,29 @@ function catavatar_addon_settings_post(App $a, &$s)
...
-		// I'm not sure it's the correct way to do this...
-		$r = dba::update('contact',
-			['photo' => $url . '/4', 'thumb' => $url . '/5', 'micro' => $url . '/6', 'avatar-date' => DateTimeFormat::utcNow()],
-			['uid' => local_user(), 'self' => 1]
-		);
-		if ($r===false) {
-			notice(L10n::t('There was an error, the cat ran away.'));
+		$self = dba::selectFirst('contact', ['id'], ['uid' => local_user(), 'self' => true]);
+		if (!DBM::is_result($self)) {
+			notice(L10n::t("The cat hadn't found itself."));
```
Comment (tobiasd) [inline catavatar/catavatar.php]: Can you please re-generate the messages.po file for the addon as you added a new string to the translation pipe.

Comment (annando) [reply catavatar/catavatar.php]: Only if you tell me, how I do this :-)

Comment (tobiasd) [reply catavatar/catavatar.php]: Go into the directory of the addon you want to extract the strings from and then call the `run_xgettext` tool from the Friendica tools with the parameter `-s`. So in this case

    cd /path/to/friendica-addons/catavatar
    /path/to/friendica/bin/run_xgettext.sh -s

should let the magic happen.

File: catavatar/catavatar.php [hunk 2]
```diff
@@ -256,10 +252,12 @@ function catavatar_content(App $a)
...
 **/
 
-function build_cat($seed='', $size=0){
-
+function build_cat($seed = '', $size = 0)
+{
 	// init random seed
-	if($seed) srand( hexdec(substr(md5($seed),0,6)) );
+	if ($seed) {
+		srand(hexdec(substr(md5($seed),0,6)));
```
Comment (tobiasd) [inline catavatar/catavatar.php]: I guess there should be spaces after the `,`

File: catavatar/catavatar.php [hunk 3]
```diff
@@ -277,22 +275,30 @@ function build_cat($seed='', $size=0){
...
 
 	// add parts
-	foreach($parts as $part => $num){
+	foreach ($parts as $part => $num){
 		$file = dirname(__FILE__).'/avatars/'.$part.'_'.$num.'.png';
 
 		$im = @imagecreatefrompng($file);
-		if(!$im) die('Failed to load '.$file);
+		if (!$im) {
+			die('Failed to load '.$file);
```
Comment (tobiasd) [inline catavatar/catavatar.php]: And around the `.` operator.

File: catavatar/lang/C/messages.po [hunk 4]
```diff
@@ -17,26 +17,34 @@ msgstr ""
...
 msgid "Cat Avatar Settings"
 msgstr ""
 
-#: catavatar.php:103 catavatar.php:112 catavatar.php:121
+#: catavatar.php:100
+msgid "The cat hadn't found itself."
+msgstr ""
+
+#: catavatar.php:109
 msgid "There was an error, the cat ran away."
```
Comment (tobiasd) [inline catavatar/lang/C/messages.po]: Wie viele Katzen werden Selbstfindungsprobleme haben? I like the humor with these cats.

**Pull request merged**